src/cov.jl

```diff
@@ -153,3 +153,37 @@ function cor2cov!(C::AbstractMatrix, s::AbstractArray)
     end
     return C
 end
+
+"""
+    CovarianceEstimator
+
+Abstract type for covariance estimators.
+"""
+abstract type CovarianceEstimator end
+
+"""
+    cov(ce::CovarianceEstimator, x::AbstractVector)
+
+Compute the sample variance of the vector `x` using estimator `ce`.
+"""
+cov(ce::CovarianceEstimator, x::AbstractVector) =
+    error("cov is not defined for $(typeof(ce)) and $(typeof(x))")
+
+
+"""
+    cov(ce::CovarianceEstimator, x::AbstractVector, y::AbstractVector)
+
+Compute the covariance of the vectors `x` and `y` using estimator `ce`.
+"""
+cov(ce::CovarianceEstimator, x::AbstractVector, y::AbstractVector) =
+    error("cov is not defined for $(typeof(ce)), $(typeof(x)) and $(typeof(y))")
+
+
+"""
+    cov(ce::CovarianceEstimator, X::AbstractMatrix, vardim::Int=1)
```

Comment:
While we're at it, `vardim` would better be a keyword argument (like `dims` in Base now). We should also change other methods (but this can be done in another PR).

Reply:
OK, I'll change it then and prepare a separate PR for other methods.

Reply:
Thanks. Actually I'm not sure whether it should be called `vardim` or `dims`. The former was probably more explicit, and we don't expect supporting multiple dimensions I guess.

Reply:
`dims` is used in Statistics stdlib, so making it consistent makes sense. For me it works either way, so I'll do whatever you decide.

Reply:
OK since `cov` in `Statistics` uses `dims` I guess we don't really have a choice.

Reply:
There is also some confusion because `cov` from `Statistics` has `dims` as a keyword argument and `StatsBase` has `vardim` that is a positional argument. I actually prefer the approach of `Statistics` as it's a bit more explicit.